client/app/reducers/selection.coffee

```diff
@@ -0,0 +1,38 @@
+Immutable = require 'immutable'
+
+{ActionTypes} = require '../constants/app_constants'
+
+DEFAULTSTATE =
+    selected: Immutable.Set()
+    allSelected: false
+
+module.exports = (state = DEFAULTSTATE, action) ->
+    switch action.type
+        when ActionTypes.MAILBOX_SELECT_ALL
+            if state.allSelected
+                nextState = DEFAULTSTATE
+            else
+                nextState =
+                    allSelected: true,
+                    selected: Immutable.Set()
+
+        when ActionTypes.MAILBOX_SELECT
```

Comment:
Same remark as for `MAILBOX_DESELECT_ALL`, what about a `MAILBOX_DESELECT` action ?